lib/skate/detours/missed_stops.ex

```diff
@@ -0,0 +1,177 @@
+alias Util.Location
+
+defmodule Skate.Detours.MissedStops do
+  @moduledoc """
+  Processes Shape and Stop data to compute contiguous stops between the
+  `connection_start` and `connection_end` parameters.
+  """
+
+  @type t :: %__MODULE__{
+          connection_start: Util.Location.From.t(),
+          connection_end: Util.Location.From.t(),
+          stops: nonempty_list(Util.Location.From.t()),
+          shape: nonempty_list(Util.Location.From.t())
+        }
+
+  @enforce_keys [:connection_start, :connection_end, :stops, :shape]
+  defstruct [:connection_start, :connection_end, :stops, :shape]
+
+  @doc """
+  Returns the contiguous list of stops, from the input parameter `cfg.stops`.
+  """
+  @spec missed_stops(cfg :: __MODULE__.t()) :: [Util.Location.From.t()]
+  def(
+    missed_stops(%__MODULE__{
+      stops: stops,
+      shape: shape,
+      connection_start: connection_start,
+      connection_end: connection_end
+    })
+  ) do
+    segmented_shape = segment_shape_by_stops(shape, stops)
+
+    {connection_start, connection_end}
+    |> missed_segments(segmented_shape)
+    |> Enum.map(& &1.stop)
+  end
+
+  @spec missed_segments(
+          {connection_start :: Util.Location.From.t(), connection_end :: Util.Location.From.t()},
+          segmented_shape :: [Skate.Detours.ShapeSegment.t()]
+        ) :: [Skate.Detours.ShapeSegment.t()]
+  defp missed_segments({connection_start, connection_end}, segmented_shape) do
+    %{index: start_index} = get_index_by_min_dist(segmented_shape, connection_start)
+
+    remaining_segments = Enum.drop(segmented_shape, start_index)
+
+    %{index: end_count} =
+      get_index_by_min_dist(remaining_segments, connection_end)
+
+    if end_count == 0 do
+      case Enum.at(segmented_shape, start_index) do
+        # As long as there's only one stop in the segment,
+        # we'll assume that we won't miss any stops
+        # because we're rejoining the same segment
+        %Skate.Detours.ShapeSegment{
+          stop: :none
+        } ->
+          []
+
+        # If there's more than one stop in this segment,
+        # Then return all the stops
+        %Skate.Detours.ShapeSegment{
+          stop: stop
+        } ->
+          [stop]
+      end
+
```

Comment:
```suggestion
```

If I'm reading this correctly, the results of this `case` statement aren't being used, since it's followed immediately by just returning `[]` (which makes sense - if the connection end-point is before the collection of segments we're currently looking at, then there shouldn't be any missed stops!)

Reply:
61b816a880a0b694b6db68acd79f4771aabb91ee

Reply:
Oh nice - that's even simpler